ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -41,6 +41,78 @@ EOF
     fi
 }
 
+# Update RHEL repo before upgrade
+function rhel_repo(){
+    echo "Updating RHEL node repo"
+    # Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+    # to be able to SSH.
+    if ! whoami &> /dev/null; then
+        if [[ -w /etc/passwd ]]; then
+            echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+        else
+            echo "/etc/passwd is not writeable, and user matching this uid is not found."
+            exit 1
+        fi
+    fi
+    SOURCE_REPO_VERSION=$(echo "${SOURCE_VERSION}" | cut -d'.' -f1,2)
+    TARGET_REPO_VERSION=$(echo "${TARGET_VERSION}" | cut -d'.' -f1,2)
+    export SOURCE_REPO_VERSION
+    export TARGET_REPO_VERSION
+
+    cat > repo.yaml <<-'EOF'
+---
+- name: Update repo Playbook
+  hosts: new_workers
+  any_errors_fatal: true
+  gather_facts: false
+  vars:
+    source_repo_version: "{{ lookup('env', 'SOURCE_REPO_VERSION') }}"
+    target_repo_version: "{{ lookup('env', 'TARGET_REPO_VERSION') }}"
+  tasks:
+  - name: Wait for host connection to ensure SSH has started
+    wait_for_connection:
+      timeout: 600
+  - name: Find all of the files inside /etc/yum.repos.d/
```

Comment:
seems we don't have to try to replace the `source_repo_version` in all the repo files, the ose repo has a fixed name as `/etc/yum.repos.d/rhel-{{ major_platform_version }}-server-ose-rpms.repo` in prow
and 
```
    platform_version: "{{ lookup('env', 'PLATFORM_VERSION') }}"
    major_platform_version: "{{ platform_version[:1] }}"
```
so that we don't want to replace a 4.x word in other repos unexpectedly, although it's low probability.

Reply:
Thanks for the suggestion! The drawback of ^ is just when we change the repo file name in the scaleup step we need to update it in the step accordingly. But it's fine with me too.